lib/matplotlib/tests/test_text.py

```diff
@@ -313,169 +313,59 @@ def test_bbox_clipping():
-    eq_(points[1, 0], 0.0)
-    eq_(points[1, 1], 50.0)
-    eq_(points[0, 1], 50.0)
-
+def test_annotation_negative_ax_coords():
+    fig, ax = plt.subplots()
 
-@image_comparison(baseline_images=['basictext_wrap'],
-                  extensions=['png'])
-def test_basic_wrap():
-    fig = plt.figure()
-    plt.axis([0, 10, 0, 10])
-    t = "This is a really long string that I'd rather have wrapped so that" \
-        " it doesn't go outside of the figure, but if it's long enough it" \
-        " will go off the top or bottom!"
-    plt.text(4, 1, t, ha='left', rotation=15, wrap=True)
-    plt.text(6, 5, t, ha='left', rotation=15, wrap=True)
-    plt.text(5, 5, t, ha='right', rotation=-15, wrap=True)
-    plt.text(5, 10, t, fontsize=18, style='oblique', ha='center',
-             va='top', wrap=True)
-    plt.text(3, 4, t, family='serif', style='italic', ha='right', wrap=True)
-    plt.text(-1, 0, t, ha='left', rotation=-15, wrap=True)
-
-
-@image_comparison(baseline_images=['fonttext_wrap'],
+    ax.annotate('+ pts',
+                xytext=[30, 20], textcoords='axes points',
+                xy=[30, 20], xycoords='axes points', fontsize=32)
+    ax.annotate('- pts',
+                xytext=[30, -20], textcoords='axes points',
+                xy=[30, -20], xycoords='axes points', fontsize=32,
+                va='top')
+    ax.annotate('+ frac',
+                xytext=[0.75, 0.05], textcoords='axes fraction',
+                xy=[0.75, 0.05], xycoords='axes fraction', fontsize=32)
+    ax.annotate('- frac',
+                xytext=[0.75, -0.05], textcoords='axes fraction',
+                xy=[0.75, -0.05], xycoords='axes fraction', fontsize=32,
+                va='top')
+
+    ax.annotate('+ pixels',
+                xytext=[160, 25], textcoords='axes pixels',
+                xy=[160, 25], xycoords='axes pixels', fontsize=32)
+    ax.annotate('- pixels',
+                xytext=[160, -25], textcoords='axes pixels',
+                xy=[160, -25], xycoords='axes pixels', fontsize=32,
+                va='top')
+
+
+@image_comparison(baseline_images=['annotation_negative_fig_coords'],
                   extensions=['png'])
-def test_font_wrap():
-    fig = plt.figure()
-    plt.axis([0, 10, 0, 10])
-    t = "This is a really long string that I'd rather have wrapped so that" \
-        " it doesn't go outside of the figure, but if it's long enough it" \
-        " will go off the top or bottom!"
-    plt.text(4, -1, t, fontsize=18, family='serif', ha='left', rotation=15,
-             wrap=True)
-    plt.text(6, 5, t, family='sans serif', ha='left', rotation=15, wrap=True)
-    plt.text(5, 5, t, weight='light', ha='right', rotation=-15, wrap=True)
-    plt.text(5, 10, t, weight='heavy', ha='center', va='top', wrap=True)
-    plt.text(3, 4, t, family='monospace', ha='right', wrap=True)
-    plt.text(-1, 0, t, fontsize=14, style='italic', ha='left', rotation=-15,
-             wrap=True)
+def test_annotation_negative_fig_coords():
```

Comment:
The [GraphicsContextBase](https://github.com/matplotlib/matplotlib/blob/master/lib/matplotlib/backend_bases.py#L769) and we simply check that the appropriate attributes are set at draw time.

It is another test completely to ensure that a backend is producing the appropriate pixels for a GraphicsContextBase, IMHO.